metabrainz/model/access_log.py

```diff
@@ -13,40 +16,53 @@ class AccessLog(db.Model):
 
     Each request needs to be logged. Logging is done to keep track of number of
     requests in a fixed time frame. If there is an unusual number of requests
-    being made in this time frame, some kind of action is taken. See
-    implementation of this model for more details.
+    being made from different IP addresses in this time frame, action is taken.
+    See implementation of this model for more details.
     """
     __tablename__ = 'access_log'
 
     token = db.Column(db.String, db.ForeignKey('token.value'), primary_key=True)
     timestamp = db.Column(db.DateTime(timezone=True), primary_key=True, default=datetime.utcnow)
+    ip_address = db.Column(postgres.INET)
 
     @classmethod
-    def create_record(cls, access_token):
+    def create_record(cls, access_token, ip_address):
         """Creates new access log record with a current timestamp.
 
-        It also checks if `HOURLY_ALERT_THRESHOLD` is exceeded, alerts admins if
-        that's the case.
+        It also checks if `DIFFERENT_IP_LIMIT` is exceeded within current time
+        and `CLEANUP_RANGE_MINUTES`, alerts admins if that's the case.
 
         Args:
             access_token: Access token used to access the API.
+            ip_address: IP access used to access the API.
 
         Returns:
             New access log record.
         """
-        new_record = cls(token=access_token)
+        new_record = cls(
+            token=access_token,
+            ip_address=ip_address,
+        )
         db.session.add(new_record)
         db.session.commit()
 
+        cls.remove_old_ip_addr_records()
+
         # Checking if HOURLY_ALERT_THRESHOLD is exceeded
-        count = cls.query.filter(cls.timestamp > datetime.now() - timedelta(hours=1),
-                                 cls.token == access_token).count()
-        if count > HOURLY_ALERT_THRESHOLD:
-            msg = "Hourly access threshold exceeded for token %s" % access_token
+        count = cls.query \
+            .distinct(cls.ip_address) \
+            .filter(cls.timestamp > datetime.now(pytz.utc) - timedelta(minutes=CLEANUP_RANGE_MINUTES),
+                    cls.token == access_token) \
+            .count()
+        if count > DIFFERENT_IP_LIMIT:
+            msg = ("Hourly access threshold exceeded for token %s\n\n"
```

Comment:
I would love the see the count in the message. Exactly how many times has it been used? (might save me a trip to the web page to check)